docs/extensions.md

```diff
@@ -248,19 +248,70 @@ Unfortunately, the two unencrypted classes are derived from the encrypted one ra
-```
+```python
 class BaseConfig():
     SWAN_API_URL="https://dev-api.swanbitcoin.com"
 
 class ProductionConfig(BaseConfig):
     SWAN_API_URL="https://api.swanbitcoin.com"
 ```
 In your code, you can access the correct value as in any other flask-code, like `api_url = app.config.get("SWAN_API_URL")`. If the instance is running a config (e.g. `DevelopmentConfig`) which is not available in your service-specific config (as above), the inheritance-hirarchy from the mainconfig will get traversed and the first hit will get get configured. In this example, it would be `BaseConfig`.
+
 ### Callback methods
-Your service class will inherit a callback-method which will get called for various reasons with the "reason" being a string as the first parameter. Checkout the `cryptoadvance.specter.services.callbacks` file for the specific callbacks.
+Your service class will inherit a callback method which will get called for various reasons with the "reason" being a string as the first parameter. Checkout the `cryptoadvance.specter.services.callbacks` file for the specific callbacks.
 
-Some important one is the `after_serverpy_init_app` which passes a `Scheduler` class which can be used to setup regular tasks. A list of currently implemented callback-methods along with their descriptions are available in [`/src/cryptoadvance/specter/services/callbacks.py`](https://github.com/cryptoadvance/specter-desktop/blob/master/src/cryptoadvance/specter/services/callbacks.py).
+Some important one is the `after_serverpy_init_app` which passes a `Scheduler` class which can be used to setup regular tasks. A list of currently implemented callback methods along with their descriptions are available in [`/src/cryptoadvance/specter/services/callbacks.py`](https://github.com/cryptoadvance/specter-desktop/blob/master/src/cryptoadvance/specter/services/callbacks.py).
 
 ### `controller.py`
-The minimal url routes for `Service` selection and management. As usualy in Flask, `templates` and `static` resources are in their respective subfolders. Please note that there is an additional directory with the id of the extension which looks redundant at first. This is due to the way blueprints are loading templates and ensures that there are no naming collisions. Maybe at a later stage, this can be used to let plugins override other plugin's templates.
+The minimal url routes for `Service` selection and management. As usual in Flask, `templates` and `static` resources are in their respective subfolders. Please note that there is an additional directory with the id of the extension which looks redundant at first. This is due to the way blueprints are loading templates and ensures that there are no naming collisions. Maybe at a later stage, this can be used to let plugins override other plugin's templates.
+
+### Extending the settings dialog
+You can extend the settings dialog with your own templates. To do that, create a callback method in your service like:
```

Comment:
It's unclear to me which settings dialog we're talking about. As I'm trying to get something to show up, I'm not sure whether I should be looking at wallet settings or the settings for my extension. Am I just expecting the tabs to change, or am I expecting the contents of one of the existing settings tabs to change? So far I'm not having luck getting something to change and I'm not sure if it's because I'm looking in the wrong place or I just implemented the changes incorrectly.

It would also be helpful to give a concrete example or two as to what sort of thing you might be able to do with this feature.

Reply:
Ah, I finally figured it out by just manually typing the route in the browser. It's the top-level setting dialog. Definitely suggest making that clearer, and maybe including a small screenshot to help the developer understand what they're doing.

Can you clarify the relationship between the settings I would define in the global settings vs. the extension settings? Is the new settings tab in the global settings also for configuring the extension?

Reply:
Ok, screenshot added, check the preview:
https://deploy-preview-1648--specter-desktop-docs.netlify.app/extensions